src/handler.cpp

```diff
@@ -1667,12 +1676,14 @@ bool equip_char(struct char_data * ch, struct obj_data * obj, int pos)
                   obj->affected[j].modifier,
                   obj->obj_flags.bitvector, TRUE);
 
-  affect_total(ch);
-  calc_weight(ch);
+  if (recalc) {
+    affect_total(ch);
+    calc_weight(ch);
```

Comment:
`calc_weight()` should always be called, it's a safeguard function against the many (many) issues we've had around weight issues in the past.